IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -73,33 +73,38 @@ - (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem
     });
 }
 
-- (void)removeStoredCredentialForKey:(MSIDDefaultCredentialCacheKey *)key
+- (void)removeStoredCredentialForKey:(MSIDCacheKey *)key
 {
-    dispatch_barrier_async(self.queue, ^{
-        NSString *credentialKey = [self getCredentialKey:key];
-        MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Removing credential for key %@.", MSID_PII_LOG_MASKABLE(credentialKey));
-        [self.cacheObjects removeObjectForKey:credentialKey];
+    dispatch_barrier_sync(self.queue, ^{
+        [self.cacheObjects removeObjectForKey:key];
     });
 }
 
-- (NSArray<MSIDCredentialCacheItem *> *)storedCredentialsForKey:(MSIDDefaultCredentialCacheKey *)key
+- (NSArray<MSIDCredentialCacheItem *> *)storedCredentialsForKey:(MSIDCacheKey *)key
 {
-    __block NSArray<MSIDCredentialCacheItem *> *credentials =  @[];
+    __block NSMutableArray<MSIDCredentialCacheItem *> *credentials =  [[NSMutableArray alloc] init];
     
     dispatch_sync(self.queue, ^{
         if (key.account && key.service)
         {
-            NSString *credentialKey = [self getCredentialKey:key];
-            MSIDCredentialCacheItem *credential = [self.cacheObjects objectForKey:credentialKey];
+            MSIDCredentialCacheItem *credential = [self.cacheObjects objectForKey:key];
             if (credential)
             {
-                credentials = @[credential];
+                [credentials addObject:credential];
             }
         }
         else
         {
-            NSArray *storedCredentials = [self.cacheObjects allValues];
-            credentials = [storedCredentials filteredArrayUsingPredicate:[self createPredicateForKey:key]];
+            NSArray *storedKeys = [self.cacheObjects allKeys];
+            NSArray *filteredKeys = [storedKeys filteredArrayUsingPredicate:[self createPredicateForKey:key]];
+            for(MSIDCacheKey *key in filteredKeys)
```

Comment:
would it be faster to enumerate dictionary including keys and values? right now you have two enumerations: one for array of keys, another to retrieve each item.

Reply:
NSPredicate works only only on NSArray, NSMutableArray, NSSet and NSMutableSet. NSDictionary can use predicates by filtering its keys or values (both NSArray objects). So leaving this as is.

Reply:
If we can optimize performance, I wouldn't lock down on nspredicate. e.g. we can enumerate with enumerateKeysAndObjectsUsingBlock and match by key ourselves without a predicate.

Reply:
I checked the performance using predicate and it is faster or very comparable in units of milliseconds. Also the code with predicate is much more readable so I think we are good here.